app.R

```diff
@@ -32,16 +60,22 @@ ui <- fluidPage(
     )
 )
 
-# Define server logic required to draw a histogram
-server <- function(input, output) {
+# Define server logic required to draw a plot
+server <- function(input, output, session) {
+
+  # Filter dataframe based on user selected genus
+  filtered <- reactive({
+    stem_spp_filter <- filter(stem_spp, Genus == toString(input$genus)) # Convert selected genus into a string
+    return(stem_spp_filter)
+  })
 
     output$distPlot <- renderPlot({
-        # generate bins based on input$bins from ui.R
-        x    <- faithful[, 2]
-        bins <- seq(min(x), max(x), length.out = input$bins + 1)
 
-        # draw the histogram with the specified number of bins
-        hist(x, breaks = bins, col = 'darkgray', border = 'white')
+      # Draw frequency polygon plot
+      ggplot(filtered(), aes(x = dbh, color = Census)) +
+        geom_freqpoly(position = "identity", binwidth = 125) +
```

Comment:
Consider
* Using a more refined "frequency" bins
* Cutting off the plot at 50cm since so few trees have dbh greater than 50

```suggestion
        geom_freqpoly(position = "identity", binwidth = 1) +
        coord_cartesian(xlim = c(0, 50)) +
```